lib/cli-engine.js

```diff
@@ -496,15 +517,19 @@ CLIEngine.prototype = {
             results.push(res);
         }
 
-        files.forEach(function(pattern) {
-            glob.sync(pattern, globOptions).forEach(executeOnFile);
+        patterns.forEach(function(pattern) {
+            if (shell.test("-f", pattern) && !ignoredPaths.contains(pattern)) {
```

Comment:
Not sure what you are asking. Can you clarify? 

This new if block I added checks if the pattern passed is actually a file and not ignored by the `.eslintignore` file. If that is the case it skips glob and directly executes the file.